documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -351,4 +353,232 @@ Control characters are not allowed in a specification.
+
+**Why the rule is important**: To reduce duplication, maintain consistent structure in ARM specifications, and ensure `subscriptionId` and `api-version` are created as client properties in the generated code.
+
+**How to fix the violation**: Ensure `subscriptionId` and `api-version` are declared in the global parameters section of the document.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2020" />R2020 RequiredPropertiesMissingInResourceModel
+**Output Message**: Model definition '{0}' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly.
+
+**Description**: Per ARM guidelines, a `Resource` model must have the `name`, `id` and `type` properties defined as `readOnly` in its hierarchy.
+
+**Why the rule is important**: `name`, `type` and `id` are readonly properties set on the service end. Also, per ARM guidelines each `Resource` type model must have these properties defined in its hierarchy.
+
+**How to fix the violation**: Ensure the `Resource` type model has the properties `name`, `type` and `id` and they are marked as `readOnly:true`. 
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S2056" />S2056 RequiredReadOnlyProperties
+**Output Message**: Property '{0}' is a required property. It should not be marked as 'readonly'.
+
+**Description**: A model property cannot be both `readOnly` and `required`. A `readOnly` property is something that the server sets when returning the model object while required is a property required to be set when sending it as a part of the request body.
+
+**Why the rule is important**: SDK generation fails when this rule is violated.
+
+**How to fix the violation**: Ensure that the given property is either marked as `readonly: true` or `required` but not both.
+
+**Bad Example**:
+```
+"MyModel": {
+  "properties":{
+    "MyProp":{
+      "type": "string",
+      "description": "sample prop",
+      "readOnly": true
+    }
+  },
+  "required": ["MyProp"]
+}
+```
+
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S2014" />S2014 SubscriptionIdParameterInOperations
```

Comment:
Is this in the above table. May be my find is not working. Let's make sure it's there :)